### PR TITLE
feat: Add a warning in case senders are undenied manually from db

### LINKED ISSUE #338: Deny sender again if indexers remove from denylist
**Describe the bug**
We want to be extra sure that people MUST NOT remove a sender from the denylist. In case they do, deny it again and send a new message saying that by doing so, they are risking to lose their query fees money.

They can always just shutdown tap-agent, but this should not be our problem.

### PR DESCRIPTION
**Pull request by carlosvdr**
This also creates a new function which does the check for inital denial of a sender
fixes #338

### CODE REVIEW

File: tap-agent/src/agent/sender_account.rs [hunk 1]
```diff
@@ -574,6 +575,19 @@ impl Actor for SenderAccount {
 
                 match receipt_fees {
                     ReceiptFees::NewReceipt(value) => {
+                        // If state is denied and receivied new receipt, sender was removed manually from DB
```
Comment (gusinacio) [inline tap-agent/src/agent/sender_account.rs:578]: typo, `received`

File: tap-agent/src/agent/sender_account.rs [hunk 2]
```diff
@@ -574,6 +575,19 @@ impl Actor for SenderAccount {
...
+                        if state.denied {
+                            tracing::warn!(
+                                "
+                                No new receipts should have been received, sender has been denied before. \
+                                You SHOULD NOT remove a denied sender manually from the database. \
+                                If you do so you are exposing yourself to potentially LOOSING ALL of your query
```
Comment (gusinacio) [inline tap-agent/src/agent/sender_account.rs:584]: typo: `LOSING`

File: tap-agent/src/agent/sender_account.rs [hunk 3]
```diff
@@ -574,6 +575,19 @@ impl Actor for SenderAccount {
...
+                                fee MONEY.
```
Comment (gusinacio) [inline tap-agent/src/agent/sender_account.rs:585]: I liked the messages, do you think some **** around the capitalized letters can make it more flashy?

**Review by gusinacio — CHANGES_REQUESTED** [review]
Really good code. just some typos

File: tap-agent/src/agent/sender_account.rs (hunk 3 above)
Comment (carlosvdr) [reply tap-agent/src/agent/sender_account.rs:585]: Okay ya

**Review by gusinacio — APPROVED** [review]
LGTM, simple and solves our issue!

**Pull request merged**